### DISCUSSION TITLE
Scroll to top of Modal?

### DISCUSSION THREAD

**Discussion by juliuslissau** [mantinedev/mantine#5590] [Q&A] [answered]
Is there a way to scroll to the top of a `Modal` with Javascript?

I've been messing around with setting a ref and doing `ref.current.scrollTop = 0` and tried using `ScrollArea` and `use-scroll-into-view`, but nothing seems to work for me.

**Comment by yverby** [comment, marked as answer]
`ScrollArea` component
```ts
const ref = useRef(null);

const scrollTop = () => ref.current?.scrollTo({ top: 0 });

<Modal
  scrollAreaComponent={(props) => (
    <ScrollArea.Autosize {...props} viewportRef={ref} />
  )}
>
  ...
</Modal>
```

Native scroll
```ts
const ref = useRef(null);

const scrollTop = () => ref.current?.scrollTo({ top: 0 });

<Modal
  scrollAreaComponent={({ style, ...props }) => (
    <div {...props} ref={ref} style={{ ...style, overflow: 'auto' }} />
  )}
>
  ...
</Modal>
```

**Reply by yverby** [reply]
@rtivital Does it make sense to add an extra property `overflow: 'auto'` to `styles` for `Scroll` in [this](https://github.com/mantinedev/mantine/blob/master/packages/%40mantine/core/src/components/Modal/ModalContent.tsx) component, so as not to add it manually? Because in this case it doesn't work quite correctly for the native scroll element.

**Reply by juliuslissau** [reply]
@yverby Thank you! But ehm...
It solves the scroll problem, but introduces another problem: 
Whenever I leave the tab and come back, the page scrolls to the top. Seems like any component inside the Modal gets unmounted and then remounted when returning to the tab

**Reply by yverby** [reply]
@juliuslissau  You can try to reproduce your issue using [codesandbox](https://codesandbox.io/p/sandbox/mantine-template-9ze89)
I am not sure if the problem is with `Modal` component, it could be something else

**Reply by juliuslissau** [reply]
@yverby I was able to re-create the issue now: https://codesandbox.io/p/sandbox/vigorous-roman-xmpmc7

Personally, in my code base, what happens is that I use "useSession()" from "next-auth/react" which seems to trigger a re-render when the tab is re-focused. And inserting your "scrollAreaComponent" code, that then re-renders the whole Modal for some reason.

I recreated the re-rendering on re-focus issue using a useEffect (code written by ChatGPT) instead of installing the library.

You have to open the codesandbox up in a full page, this link should work: https://xmpmc7.csb.app/

Then open the modal, scroll to the bottom, open a new tab, and then go back to the old tab again. You will see the modal has scrolled to the top.

**Reply by juliuslissau** [reply]
@rtivital I just noticed that the above issue also happens with the Accordion component. Meaning if you have an open Acoordion.Item, and then opens another tab, and then re-focuses the old tab, then the Accordion.Item will close. 
So this might be a (small) general issue with Mantine?

**Reply by yverby** [reply]
@juliuslissau  Try wrapping the function that is passed as `scrollAreaComponent` prop in `useCallback`

**Reply by juliuslissau** [reply]
@yverby Perfect, that fixes the issue! Thank you so much!
